File: app/src/main/java/com/bigscreen/mangindo/network/NetworkError.java
package com.bigscreen.mangindo.network;


import java.io.IOException;

import retrofit2.adapter.rxjava.HttpException;

public class NetworkError {

    private static final int ERROR_UNKNOWN = 0;
    private static final int ERROR_SERVER = 1;
    private static final int ERROR_CONNECTION = 2;

    public static final String MESSAGE_ERROR = "Something went wrong. Please try again.";
    private static final String MESSAGE_ERROR_SERVER = "Server error. Please try again.";
    private static final String MESSAGE_ERROR_CONNECTION = "Connection error. Please Try again.";

    private Throwable throwable;

    public NetworkError(Throwable throwable) {
        this.throwable = throwable;
    }

    public String getErrorMessage() {
        switch (getErrorType()) {
            case ERROR_SERVER:
                return MESSAGE_ERROR_SERVER;
            case ERROR_CONNECTION:
                return MESSAGE_ERROR_CONNECTION;
            default:
                return throwable.getMessage();
        }
    }

    private int getErrorType() {
        if (throwable instanceof HttpException)
            return ERROR_SERVER;
        else if (throwable instanceof IOException)
            return ERROR_CONNECTION;
        else
            return ERROR_UNKNOWN;
    }
}

File: app/src/main/java/com/bigscreen/mangindo/network/NetworkError.kt
package com.bigscreen.mangindo.network

import retrofit2.HttpException
import java.io.IOException

class NetworkError(private val throwable: Throwable) {

    companion object {
        private const val ERROR_UNKNOWN = 0
        private const val ERROR_SERVER = 1
        private const val ERROR_CONNECTION = 2
        private const val MESSAGE_ERROR_SERVER = "Server error. Please try again."
        private const val MESSAGE_ERROR_CONNECTION = "Connection error. Please Try again."
        const val MESSAGE_ERROR = "Something went wrong. Please try again."
    }

    fun getErrorMessage() = when (getErrorType()) {
        ERROR_SERVER -> MESSAGE_ERROR_SERVER
        ERROR_CONNECTION -> MESSAGE_ERROR_CONNECTION
        else -> throwable.message.orEmpty()
    }

    private fun getErrorType() = when (throwable) {
        is HttpException -> ERROR_SERVER
        is IOException -> ERROR_CONNECTION
        else -> ERROR_UNKNOWN
    }
}
